File: Sif/src/app/core/models/comment.ts

import { User } from '@auth0/auth0-angular';

export type CommentStatus = 'new' | 'released' | 'spam' | 'trash'
export class Comment {
    commentId?: number;
    userMail?: string;
    commentText?: string;
    articleTitle?: string;
    articleId?: number;
    creationDate?: Date;
    answerOf?: number;
    replies?: Comment[];
    userId?: string;
    user?: User;
    status?: CommentStatus;
}

File: Thor/Models/Comment.cs
using System;
using System.Collections.Generic;

namespace Thor.Models
{
  public class Comment
  {
    public int CommentId { get; set; }
    public int ArticleId { get; set; }
    public int? AnswerOf { get; set; }
    public string CommentText { get; set; }
    public DateTime CreationDate { get; set; }
    public string Status { get; set; }
    public string UserName { get; set; }
    public string UserId { get; set; }
    public List<Comment> Answers { get; set; }
  }
}